src/main/java/redis/clients/jedis/commands/MultiKeyBinaryJedisClusterCommands.java

```diff
@@ -1,144 +1,63 @@
-
-  Long zunionstore(byte[] dstkey, ZParams params, byte[]... sets);
-
-  byte[] brpoplpush(byte[] source, byte[] destination, int timeout);
-
-  Long publish(byte[] channel, byte[] message);
-
-  void subscribe(BinaryJedisPubSub jedisPubSub, byte[]... channels);
-
-  void psubscribe(BinaryJedisPubSub jedisPubSub, byte[]... patterns);
-
-  Long bitop(BitOP op, byte[] destKey, byte[]... srcKeys);
-
-  String pfmerge(byte[] destkey, byte[]... sourcekeys);
-
-  Long pfcount(byte[]... keys);
-
-  Long touch(byte[]... keys);
-
-  ScanResult<byte[]> scan(byte[] cursor, ScanParams params);
-
-  Set<byte[]> keys(byte[] pattern);
+/**
+ * @deprecated This interface will be removed in future. Use {@link MultiKeyBinaryCommands}.
+ */
+@Deprecated
+public interface MultiKeyBinaryJedisClusterCommands extends MultiKeyBinaryCommands {
 
   /**
-   * @deprecated This method will be removed due to bug regarding {@code block} param. Use
-   * {@link #xread(redis.clients.jedis.params.XReadParams, java.util.Map.Entry...)}.
+   * @throws UnsupportedOperationException Use {@link #copy(byte[], byte[], boolean)}.
    */
-  @Deprecated
-  List<byte[]> xread(int count, long block, Map<byte[], byte[]> streams);
+  @Override
+  default Boolean copy(byte[] srcKey, byte[] dstKey, int db, boolean replace) {
+    throw new UnsupportedOperationException("Cluster mode does not support databse operations.");
+  }
 
-  List<byte[]> xread(XReadParams xReadParams, Entry<byte[], byte[]>... streams);
+  /**
+   * @throws UnsupportedOperationException Use {@link #blpop(double, byte[]...)} or
+   * {@link #blpop(int, byte[]...)}.
+   */
+  @Override
+  default List<byte[]> blpop(byte[]... args) {
+    throw new UnsupportedOperationException();
+  }
 
   /**
-   * @deprecated This method will be removed due to bug regarding {@code block} param. Use
-   * {@link #xreadGroup(byte..., byte..., redis.clients.jedis.params.XReadGroupParams, java.util.Map.Entry...)}.
+   * @throws UnsupportedOperationException Use {@link #brpop(double, byte[]...)} or{
+   * {@link #brpop(int, byte[]...)}.
    */
-  @Deprecated
-  List<byte[]> xreadGroup(byte[] groupname, byte[] consumer, int count, long block, boolean noAck,
-      Map<byte[], byte[]> streams);
+  @Override
+  default List<byte[]> brpop(byte[]... args) {
+    throw new UnsupportedOperationException();
+  }
 
-  List<byte[]> xreadGroup(byte[] groupname, byte[] consumer, XReadGroupParams xReadGroupParams,
-      Entry<byte[], byte[]>... streams);
+  /**
+   * @throws UnsupportedOperationException
+   */
+  @Override
+  default String watch(byte[]... keys) {
+    throw new UnsupportedOperationException();
+  }
 
-  Long georadiusStore(byte[] key, double longitude, double latitude, double radius, GeoUnit unit,
-      GeoRadiusParam param, GeoRadiusStoreParam storeParam);
+  /**
+   * @throws UnsupportedOperationException
+   */
+  @Override
+  default byte[] randomBinaryKey() {
+    throw new UnsupportedOperationException();
+  }
```

Comment:
Let's add messages to all these unsupported exceptions

Reply:
@mina-asham I'm temporarily marking the "unsupported exception message" comments are resolved just for easier browsing. Hope you won't mind :)